### ISSUE TITLE
Rename `CatalogKind.Processor` to `CatalogKind.Model`

### ISSUE THREAD

**Issue by igarashitm** [KaotoIO/kaoto-next#519]
### Please describe the task that needs to be done

Now that we have `CatalogKind.Pattern` which is a processors catalog, and the `CatalogKind.Processor` is actually a whole model catalog, better to rename it to `CatalogKind.Model` to avoid confusion. IDE should do it automatically, but better to be in a dedicated PR.

**Comment by igarashitm** [comment]
Nope not now, `CatalogKind.Processor` is still in use for catalog view and not trivial to fix all.

**Comment by igarashitm** [comment]
Blocked by #785

**Comment by lordrip** [comment]
Since we offer the Camel Catalog as part of the Kaoto UI, we need to keep this catalog. Now, only what needs to have a JSON schema is in the `patterns` catalog. The `model` catalog is only used for catalog viewing.
